### ISSUE TITLE
Unable to communicate with self-hosted Server

### ISSUE THREAD

**Issue by l0nax** [atuinsh/atuin#1142]
#### Description

I deployed `atuin` with docker-compose behind Caddy. Thus, a Let's Encrypt certificate is being used.

If I try to execute the `atuin register` command, I receive the following error stating that the server certificate is not trusted:
```
Error: error sending request for URL (https://atuin.l0nax.org/user/l0nax): error trying to connect: invalid peer certificate: UnknownIssuer

Caused by:
   0: error trying to connect: invalid peer certificate: UnknownIssuer
   1: invalid peer certificate: UnknownIssuer
```

* Atuin Server Version: 15.0.0 (tested with `main` too)
* Atuin Client Version: 15.0.0 (tested with custom build from `main` branch too)
* Distro: Arch (host CA bundle is up-to-date)

Executing `curl -i https://atuin.l0nax.org/user/l0nax` returns the server response as expected.

**Comment by desolatorxxl** [comment]
I had the same issue because I used an old version (17.0.0), upgrade to 18.0.2 fixed it for me.

**Comment by l0nax** [comment]
I updated the server to _18.0.2_ and the CLI:
```bash
~$ atuin --version
atuin 18.0.2
```

And if I try to execute `atuin register [...]` it still throws the unknown issuer error:
```bash
~$ atuin register -u [...] -e [...]
Registering for an Atuin Sync account
Please enter password:
Error: error sending request for url (https://atuin.l0nax.org/user/[...]): error trying to connect: invalid peer certificate: UnknownIssuer

Caused by:
   0: error trying to connect: invalid peer certificate: UnknownIssuer
   1: invalid peer certificate: UnknownIssuer

Location:
    /usr/src/debug/atuin/atuin-18.0.2/atuin-client/src/api_client.rs:49:16
```

But the certificate is valid on my server:
```bash
~$ curl https://atuin.l0nax.org 
{"homage":"\"Through the fathomless deeps of space swims the star turtle Great A'Tuin, bearing on its back the four giant elephants who carry on their shoulders the mass of the Discworld.\" -- Sir Terry Pratchett","version":"18.0.2","total_history":-1}
```

**Comment by deepanchal** [comment]
I am having a similar issue but with different error message. I have atuin running on my pi using docker.

```sh
❯ curl http://beacon.local:8888/sync/status
{"reason":"missing authorization header"}                                                                       

❯ atuin status
Error: error sending request for url (http://beacon.local:8888/sync/status): error trying to connect: operation timed out

Caused by:
   0: error trying to connect: operation timed out
   1: operation timed out

Location:
    /build/source/atuin-client/src/api_client.rs:225:20
```

**Comment by ellie** [comment]
@deepanchal could you please open your own issue? Lots of similar-but-different errors make an issue thread very hard to follow, especially when they all have different setups and root causes

**Comment by deepanchal** [comment]
@ellie Will do :+1: 
Thanks!

**Comment by SN4K3D** [comment]
Hello, i have the same error of @l0nax : 
`**Error: error sending request for url (https://atuin.toto.lan/user/toto): error trying to connect: invalid peer certificate: NotValidForName

Caused by:
   0: error trying to connect: invalid peer certificate: NotValidForName
   1: invalid peer certificate: NotValidForName

Location:
    /home/runner/work/atuin/atuin/crates/atuin-client/src/api_client.rs:50:16**`

i need use atuin with insecure cert verification like a curl -k because when i run curl -k atuin.toto.lan i have the good response:
{"homage":"\"Through the fathomless deeps of space swims the star turtle Great A'Tuin, bearing on its back the four giant elephants who carry on their shoulders the mass of the Discworld.\" -- Sir Terry Pratchett","version":"18.4.0","total_history":-1}

**Comment by cmdln** [comment]
I am trying to set up the server using docker compose and am encountering the same issue. I also have a reverse proxy with a Let's Encrypt certificate. The same certificate is valid and works with other services. I am running v18.5.0 both on the server and the CLI.

```
❯ atuin sync
Error: a request to the sync server failed: "error sending request for url (https://atuin.cmdln.net/api/v0/record): error trying to connect: invalid peer certificate: UnknownIssuer"

Location:
    crates/atuin/src/command/client/sync.rs:91:38
```

If I change to directly sync to port 8888 without TLS it works fine.

**Comment by justinmayer** [comment]
I get the following error when running `atuin register` to connect to a Atuin server powered by Docker Compose and running behind Traefik (Traefik automatically generates a proper TLS certificate via LetsEncrypt):

    Error: error sending request for url (https://mydomain.example.com/user/myuser): error trying to connect: invalid peer certificate: UnknownIssuer

    Caused by:
    0: error trying to connect: invalid peer certificate: UnknownIssuer
    1: invalid peer certificate: UnknownIssuer

    Location:
        /private/tmp/atuin-20250512-7558-ej19jg/atuin-18.6.1/crates/atuin-client/src/api_client.rs:50:16

------

The same combination of domain and certificate works just fine when loaded in a browser or via `curl`.

Environment:

- Atuin Server Version: 18.6.1
- Atuin Client Version: 18.6.1
- Client tested on Ubuntu Linux 22.04 and MacOS Sequoia 15.5

Possibly related issue: https://github.com/astral-sh/uv/issues/1819

I don’t know if the problem lies with [Reqwest](https://github.com/search?q=repo%3Aseanmonstar%2Freqwest%20invalid%20peer%20certificate&type=issues) or somewhere else.

Potential ideas to evaluate:

- [It looks like `rustls-native-certs` in Cargo.lock is at 0.6.3.](https://github.com/atuinsh/atuin/blob/58e307bc6df66f52509c57537e68194ce6d33577/Cargo.lock#L3830) Perhaps we could try a more up-to-date version?
- Perhaps we could try upgrading Reqwest from 0.11.x to to 0.12.x? https://github.com/seanmonstar/reqwest/releases
- Perhaps we could [add an `--insecure` option](https://github.com/seanmonstar/reqwest/issues/776) to work around these kinds of problems?

@Ellie: Is there anything I can do to help provide more information or help move this forward in any way?

**Comment by justinmayer** [comment]
Hi @Ellie. I have tested with both Atuin client and server on the latest version (18.10.0) and still observe the same error. What can I do to help resolve this issue?

<details><summary><strong><i>Tap to see traceback with DEBUG enabled</strong></i> (domain, IP, and username redacted):</summary>

```
Registering for an Atuin Sync account
Please enter password:
2025-10-25T07:10:20.817363Z DEBUG reqwest::connect: starting new connection: https://example.com/
2025-10-25T07:10:20.817615Z DEBUG hyper::client::connect::dns: resolving host="example.com"
2025-10-25T07:10:20.893966Z DEBUG hyper::client::connect::http: connecting to 99.99.99.99:443
2025-10-25T07:10:23.641841Z DEBUG hyper::client::connect::http: connected to 99.99.99.99:443
2025-10-25T07:10:23.641915Z DEBUG rustls::client::hs: No cached session for DnsName("example.com")
2025-10-25T07:10:23.642021Z DEBUG rustls::client::hs: Not resuming any session
2025-10-25T07:10:23.696850Z DEBUG rustls::client::hs: Using ciphersuite TLS13_AES_128_GCM_SHA256
2025-10-25T07:10:23.696866Z DEBUG rustls::client::tls13: Not resuming
2025-10-25T07:10:23.696936Z DEBUG rustls::client::tls13: TLS1.3 encrypted extensions: [Protocols([ProtocolName(6832)])]
2025-10-25T07:10:23.696941Z DEBUG rustls::client::hs: ALPN protocol is Some(b"h2")
Error: error sending request for url (https://example.com/user/myusername): error trying to connect: invalid peer certificate: UnknownIssuer

Caused by:
   0: error trying to connect: invalid peer certificate: UnknownIssuer
   1: invalid peer certificate: UnknownIssuer

Location:
    /private/tmp/atuin-20251021-8499-gsx8c7/atuin-18.10.0/crates/atuin-client/src/api_client.rs:69:16
```

</details>

**Comment by justinmayer** [comment]
@Ellie / @BinaryMuse: Do you think we could try switching Atuin from `rustls` to `native-tls` for TLS certificate handling, similar to the topic from https://github.com/atuinsh/desktop/pull/151 / https://github.com/atuinsh/desktop/issues/103? Perhaps that could also address this issue?
